### PR TITLE
better docs for make binaries

### LINKED ISSUE #689: Documentation for building the command-line client is inadequate
The current instructions at https://github.com/docker/notary/blob/master/README.md say:

> Prerequisites:
> 
> Go >= 1.6.1
> godep installed
> libtool development headers installed
> Ubuntu: apt-get install libtool-dev
> CentOS/RedHat: yum install libtool-ltdl-devel
> Mac OS (Homebrew): brew install libtool
> Run make binaries, which creates the Notary Client CLI binary at bin/notary.

I've tried to follow all these instructions, yet "make binaries" fails with dependency errors:

> GOPATH=/root/go PATH=$PATH:/usr/local/go/bin go get github.com/tools/godep
> GOPATH=/root/go:/root/notary/Godeps/_workspace PATH=$PATH:/usr/local/go/bin make binaries
> pkcs11 import was not found anywhere without a build tag, yay
> xargs: go: terminated by signal 13
> + /root/notary/bin/notary-server
> cmd/notary-server/config.go:9:2: cannot find package "github.com/Sirupsen/logrus" in any of:
> /usr/local/go/src/github.com/Sirupsen/logrus (from $GOROOT)
> /root/go/src/github.com/Sirupsen/logrus (from $GOPATH)
> /root/go/src/github.com/Sirupsen/logrus
> /root/notary/Godeps/_workspace/src/github.com/Sirupsen/logrus

Could someone update the docs with the full list of requirements?  Thanks.

### PR DESCRIPTION
**Pull request by riyazdf**
Add explicit instructions for the Go directory structure assumed for `make binaries`

Closes #689 

Signed-off-by: Riyaz Faizullabhoy <riyaz.faizullabhoy@docker.com>

### CODE REVIEW

**Comment by cyli** [comment]
LGTM :)  Thank you for fixing!

File: README.md [hunk 1]
```diff
@@ -88,3 +88,13 @@ Prerequisites:
     - Mac OS ([Homebrew](http://brew.sh/)): `brew install libtool`
 
 Run `make binaries`, which creates the Notary Client CLI binary at `bin/notary`.
+Note that `make binaries` assumes a standard Go directory structure, in which
+Notary is checked out to the `src` directory in your `GOPATH`. For example:
+```
+GOPATH=$(<path to go/>)
+go/
```
Comment (endophage) [inline README.md]: These first 2 lines are a little confusing, I'd just collapse them into 
```
$GOPATH/
    src/
        ...
```

Comment (riyazdf) [inline README.md]: great idea!  Will update.

Comment (riyazdf) [inline README.md]: Done.

**Comment by endophage** [comment]
LGTM

**Pull request merged**